File: erlang/raindrops/src/raindrops.erl
-module(raindrops).

-export([convert/1]).

-define(Factors, [{3, "Pling"}, {5, "Plang"}, {7, "Plong"}]).

convert(Number) ->
  case lists:foldl(fun({Factor, Sound}, Acc) ->
    case Number rem Factor of
      0 -> Acc ++ Sound;
      _ -> Acc
    end end, "", ?Factors) of
    "" -> integer_to_list(Number);
    Sound -> Sound
  end.

File: php/raindrops/Raindrops.php
<?php

const DICT = [
    3 => 'Pling',
    5 => 'Plang',
    7 => 'Plong',
];

function raindrops(int $n): string
{
    $r = '';
    foreach (DICT as $key => $sound) {
        if ($n % $key === 0) {
            $r .= $sound;
        }
    }
    return $r ?: (string) $n;
}
